fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/PatternForLikeValue.java

```diff
@@ -83,14 +100,18 @@ public <M extends Message> String eval(@Nonnull final FDBRecordStoreBase<M> stor
         if (patternStr == null) {
             return null;
         }
-        String[] search = SEARCH;
-        String[] replace = REPLACE;
-        if (escapeChar != null) {
+        Map<String, String> replaceMap;
+        if (escapeChar == null) {
+            replaceMap = REPLACE_MAP;
+        } else {
             SemanticException.check(escapeChar.length() == 1, SemanticException.ErrorCode.ESCAPE_CHAR_OF_LIKE_OPERATOR_IS_NOT_SINGLE_CHAR);
-            search = ObjectArrays.concat(new String[] {escapeChar + "_", escapeChar + "%"}, SEARCH, String.class);
-            replace = ObjectArrays.concat(new String[] {"_", "%"}, REPLACE, String.class);
+            replaceMap = ImmutableMap.<String, String>builderWithExpectedSize(REPLACE_MAP.size() + 2)
+                    .put(escapeChar + "_", "_")
+                    .put(escapeChar + "%", "%")
+                    .putAll(REPLACE_MAP)
+                    .build();
```

Comment:
This preserves the original iteration order, if we want to use a `replaceEach` variant that takes that into account. As it is, it would only be relevant if the escape character were somehow one of the elements in the static `REPLACE_MAP` (most probably `\\`), which would actually be equivalent to the "choose the longer search string if two at an index" rule used by the new `replaceEach`